ledger-api/grpc-definitions/com/daml/ledger/api/v1/events_service.proto

```diff
@@ -0,0 +1,364 @@
+    string transaction_log_offset = 1;
+
+    // TODO: what kind of type do we need to support multi-domain transactions?
+    // Could the below work?
+    TransactionGraphOffset transaction_graph_offset = 2;
+  }
+}
+
+message TransactionGraphOffset {
+  // Party-id to record_time vector clock.
+  map<string, uint64> vector_clock = 1;
+}
+
+// A participant-wide address of an event and the state it may depend on.
+// Comparisons are done in lexicographic order of the fields.
+message EventOffset {
+  TransactionOffset transaction_offset = 1;
+  LocalEventOffset local_event_offset = 2;
+}
+
+// A transaction-local address of an event.
+message LocalEventOffset {
+  // Equivalent to v1 node-id within a transaction.
+  // TODO: do we need something more powerful?
+  int32 event_index = 1;
+}
+
+
+// Snapshots
+////////////
+
+message GetRecentSnapshotOffsetRequest {}
+
+message GetRecentSnapshotOffsetResponse {
+  // The address of the last transaction included in the snapshot.
+  TransactionOffset snapshot_transaction_offset = 1;
+}
+
+
+// Event Streaming
+//////////////////
+
+message StreamEventsRequest {
+  // The address of the last transaction included in the snapshot.
+  // Retrieved from `GetRecentSnapshotOffset` or from client-side
+  // storage in case of resuming a stream.
+  //
+  // Required.
+  TransactionOffset snapshot_event_offset = 1;
+
+  // Optional offset after which to serve the first event, which can
+  // be set to resume streaming events.
+  EventOffset after_event_offset = 2;
+
+  // Projection determining what kind of events to serve with what data.
+  EventProjection projection = 3; 
+
+  // Filter determining which of the projected events to serve.
+  // Required
+  // 
+  // TODO: expand filter or projection to select domains to consider.
+  TransactionFilter filter = 4;
+
+}
+
+// The server guarantees to provide at least one `StreamEventsResponse` 
+// every 100ms after ingesting new events from the subscribed domains.
+message StreamEventsResponse {
+  // The offset of this event, which is strictly monotonically increasing
+  // for consecutive events wrt the order on `EventOffset`s.
+  //
+  // Thus all events from the same transaction will be served consecutively
+  // and be assigned the same `TransactionOffset`.
+  EventOffset event_offset = 1;
+
+  // The event at this offset provided the event matches the filter.
+  Event event = 2;
+}
+
+// Event Projection
+///////////////////
+
+// If no projection is set for a specific kind of event, then none of these events will be served.
+//
+// Use the default `EventProjection` to subscribe to just the changes of the ledger end.
+//
+// Note that the projections generally provide an opinionated set of of fields only.
+message EventProjection {
+  TransactionCommittedEventProjection transaction_committed_event_projection = 1;
+  ContractCreatedEventProjection contract_created_event_projection = 2;
+  ChoiceExercisedEventProjection exercised_event_projection = 4;
```

Comment:
What about `fetch`/`lookupByKey` nodes? What about other kinds of events like party allocation, package upload and vetting?